File: frappe_risk_management/kertas_kerja/report/sasaran/sasaran.js
// Copyright (c) 2022, Muhamad Jafar Sidik and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Sasaran"] = {
	"filters": [
		{
            'fieldname': 'id_kajian_risiko',
            'label': __('Kajian Risiko'),
            'fieldtype': 'Link',
            'options': 'Kajian Risiko'
        },
		{
            'fieldname': 'organisasi',
            'label': __('Organisasi'),
            'fieldtype': 'Select',
			'options': [
				'Holding',
				'Sub Holding',
				'Anak Perusahaan',
				'Cucu Perusahaan',
				'Cicit Perusahaan'
            ],
            'default':'Sub Holding'
        },
		{
            'fieldname': 'unit_kerja',
            'label': __('Unit Kerja'),
            'fieldtype': 'Link',
            'options': 'Unit Kerja'
        },
		{
            'fieldname': 'direktorat_pemilik_sasaran',
            'label': __('Direktorat'),
            'fieldtype': 'Link',
            'options': 'Direktorat'
        },
		{
            'fieldname': 'divisi_pemilik_sasaran',
            'label': __('Divisi'),
            'fieldtype': 'Link',
            'options': 'Divisi'
        },
	]
};

File: frappe_risk_management/kertas_kerja/report/sasaran/sasaran.py
# Copyright (c) 2022, Muhamad Jafar Sidik and contributors
# For license information, please see license.txt
from __future__ import unicode_literals
from contextlib import nullcontext
from frappe import _
import frappe

def execute(filters=None):
	columns =[
		{'fieldname':'name','label':'ID','fieldtype':'Data'},
		{'fieldname':'id_kajian_risiko','label': _('Kajian Risiko'),'fieldtype': 'Link','options': 'Kajian Risiko'},
		{'fieldname':'organisasi','label': 'Organisasi','fieldtype':"Data"},
		{'fieldname':'unit_kerja','label': 'Unit Kerja','fieldtype':"Link",'options': 'Unit Kerja'},
		{'fieldname':'direktorat_pemilik_sasaran','label': 'Unit Kerja Pemilik Sasaran','fieldtype':"Link",'options': 'Direktorat'},
		{'fieldname':'deskripsi_sasaran','Deskripsi Sasaran': 'Deskripsi Sasarn','fieldtype':"Data"},
	]
	data = frappe.db.get_list('Sasaran',
			filters,
			fields=['name','id_kajian_risiko','organisasi','unit_kerja','direktorat_pemilik_sasaran','deskripsi_sasaran']
		)
	frappe.msgprint("<span style='color:Red;'>Once this popup has served it's purpose, then comment out it's invocation, viz. #frappe.msgprint...</span><br><br>" + "<pre>{}</pre>".format(frappe.as_json(data)))
	return columns, data